src/Idris/Parser.hs

```diff
@@ -937,52 +937,52 @@ interface_ syn = do (doc, argDocs, acc)
 {-| Parses an interface implementation declaration
 
 @
-  Instance ::=
-    DocComment_t? 'instance' InstanceName? ConstraintList? Name SimpleExpr* InstanceBlock?
+  Implementation ::=
+    DocComment_t? 'implementation' ImplementationName? ConstraintList? Name SimpleExpr* ImplementationBlock?
     ;
 @
 
 @
-InstanceName ::= '[' Name ']';
+ImplementationName ::= '[' Name ']';
 @
 -}
-instance_ :: Bool -> SyntaxInfo -> IdrisParser [PDecl]
-instance_ kwopt syn
+implementation_ :: Bool -> SyntaxInfo -> IdrisParser [PDecl]
```

Comment:
I think this doesn’t need an underscore anymore; that was there because `instance` is a Haskell keyword.